test/controllers/records_controller_test.rb

```diff
@@ -121,6 +121,22 @@ class RecordsControllerTest < ActionController::TestCase
       assert_redirected_to collection_path(id: 1, record_id: record.id)
     end
 
+    it 'displays a failure message when attempting to associate a granule to a collection record that has been closed.' do
+      user = User.find_by(email: 'abaker@element84.com')
+      sign_in(user)
+      stub_urs_access(user.uid, user.access_token, user.refresh_token)
+
+      Record.any_instance.stubs(close!: true)
+      record = Record.find(1)
+
+      Record.stubs(:associate_granule_to_collection).returns(false)
+
+      post :associate_granule_to_collection, params: { id: record.id, associated_granule_value: false }
+
+      assert_equal 'An error occurred associating granule to the collection', flash[:notice]
```

Comment:
after giving it a closed collection record id and a assoc granule record id, the controller will run those lines of code and should return the flash message